### ISSUE TITLE
软件崩溃

### ISSUE THREAD

**Issue by kiongweidong** [shadowsocks/ShadowsocksX-NG#680]
**System and Shadowsocksx-NG version:**

- macOS 10.13.2 (16E????)
- Shadowsocksx-NG 1.7.0?

**Expected behavior:**

导入配置文件更新节点信息

**Actual behavior:**

软件崩溃直接退出

**Steps to reproduce:**

只要导入配置文件就崩溃

**How often does this happen?**

每次

**ss-local.log**

Please upload the ss-local.log file here the file is in `~/Library/Logs`
1) Open 'Advanced Settings -> enable Verbose Mode'
2) Continue run `Shadowsocksx-NG` for 5 minutes
3) Upload the `~/Library/Logs/ss-local.log` here (with or without compress)
2017-12-29 18:40:35 INFO: initializing ciphers... chacha20
 2017-12-29 18:40:35 INFO: tcp port reuse enabled
 2017-12-29 18:40:35 INFO: listening at 127.0.0.1:1086
 2017-12-29 18:43:53 ERROR: remote_recv_cb_send: Broken pipe
 2017-12-31 21:03:00 INFO: initializing ciphers... chacha20
 2017-12-31 21:03:01 INFO: tcp port reuse enabled
 2017-12-31 21:03:01 INFO: listening at 127.0.0.1:1086
 2017-12-31 23:21:17 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:18 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:38 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:38 ERROR: remote_recv_cb_recv: Connection reset by peer
 2018-01-01 14:10:41 ERROR: connect: Network is down
 2018-01-01 19:16:57 ERROR: connect: Network is unreachable
 2018-01-01 19:17:00 ERROR: remote_recv_cb_recv: Connection reset by peer
 2018-01-01 22:47:21 INFO: initializing ciphers... chacha20
 2018-01-01 22:47:21 INFO: tcp port reuse enabled
 2018-01-01 22:47:21 INFO: listening at 127.0.0.1:1086
 2018-01-02 11:31:11 ERROR: remote_recv_cb_recv: Network is down
 2018-01-02 17:21:31 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:34 ERROR: connect: Network is unreachable
 2018-01-02 17:21:34 ERROR: connect: Network is unreachable
 2018-01-02 17:21:34 ERROR: getpeername: Invalid argument
 2018-01-02 17:22:11 ERROR: connect: Network is down
 2018-01-02 17:22:11 ERROR: connect: Network is down
 2018-01-02 18:57:42 ERROR: connect: Network is unreachable
 2018-01-02 20:26:00 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 20:39:29 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 20:56:00 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 22:12:08 INFO: initializing ciphers... chacha20
 2018-01-02 22:12:08 INFO: tcp port reuse enabled
 2018-01-02 22:12:08 INFO: listening at 127.0.0.1:1086
 2018-01-03 08:20:20 ERROR: remote_recv_cb_recv: Operation timed out
 2018-01-03 23:10:09 INFO: initializing ciphers... chacha20
 2018-01-03 23:10:09 INFO: tcp port reuse enabled
 2018-01-03 23:10:09 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:11:10 INFO: initializing ciphers... rc4-md5
 2018-01-04 00:11:10 INFO: tcp port reuse enabled
 2018-01-04 00:11:10 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:14:13 INFO: initializing ciphers... aes-128-ctr
 2018-01-04 00:14:16 ERROR: failed to resolve server name, wait 2 seconds
 2018-01-04 00:14:20 ERROR: failed to resolve server name, wait 4 seconds
 2018-01-04 00:14:28 ERROR: failed to resolve server name, wait 8 seconds
 2018-01-04 00:14:44 ERROR: failed to resolve server name, wait 16 seconds
 2018-01-04 00:15:16 ERROR: failed to resolve server name, wait 32 seconds
 2018-01-04 00:16:20 ERROR: failed to resolve server name, wait 64 seconds
 2018-01-04 00:17:53 INFO: initializing ciphers... rc4-md5
 2018-01-04 00:17:53 INFO: tcp port reuse enabled
 2018-01-04 00:17:53 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:29:40 ERROR: remote_recv_cb_send: Broken pipe

**Application log**

Open the `Console.app` and search `Shadowsocksx-NG`
Copy paste the log here

**Crash Log**

If the app crashes and pop up a crash log, please copy and paste here

2017-12-29 18:40:35 INFO: initializing ciphers... chacha20
 2017-12-29 18:40:35 INFO: tcp port reuse enabled
 2017-12-29 18:40:35 INFO: listening at 127.0.0.1:1086
 2017-12-29 18:43:53 ERROR: remote_recv_cb_send: Broken pipe
 2017-12-31 21:03:00 INFO: initializing ciphers... chacha20
 2017-12-31 21:03:01 INFO: tcp port reuse enabled
 2017-12-31 21:03:01 INFO: listening at 127.0.0.1:1086
 2017-12-31 23:21:17 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:18 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:38 ERROR: remote_recv_cb_recv: Connection reset by peer
 2017-12-31 23:21:38 ERROR: remote_recv_cb_recv: Connection reset by peer
 2018-01-01 14:10:41 ERROR: connect: Network is down
 2018-01-01 19:16:57 ERROR: connect: Network is unreachable
 2018-01-01 19:17:00 ERROR: remote_recv_cb_recv: Connection reset by peer
 2018-01-01 22:47:21 INFO: initializing ciphers... chacha20
 2018-01-01 22:47:21 INFO: tcp port reuse enabled
 2018-01-01 22:47:21 INFO: listening at 127.0.0.1:1086
 2018-01-02 11:31:11 ERROR: remote_recv_cb_recv: Network is down
 2018-01-02 17:21:31 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:32 ERROR: getpeername: Invalid argument
 2018-01-02 17:21:34 ERROR: connect: Network is unreachable
 2018-01-02 17:21:34 ERROR: connect: Network is unreachable
 2018-01-02 17:21:34 ERROR: getpeername: Invalid argument
 2018-01-02 17:22:11 ERROR: connect: Network is down
 2018-01-02 17:22:11 ERROR: connect: Network is down
 2018-01-02 18:57:42 ERROR: connect: Network is unreachable
 2018-01-02 20:26:00 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 20:39:29 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 20:56:00 ERROR: remote_recv_cb_send: Broken pipe
 2018-01-02 22:12:08 INFO: initializing ciphers... chacha20
 2018-01-02 22:12:08 INFO: tcp port reuse enabled
 2018-01-02 22:12:08 INFO: listening at 127.0.0.1:1086
 2018-01-03 08:20:20 ERROR: remote_recv_cb_recv: Operation timed out
 2018-01-03 23:10:09 INFO: initializing ciphers... chacha20
 2018-01-03 23:10:09 INFO: tcp port reuse enabled
 2018-01-03 23:10:09 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:11:10 INFO: initializing ciphers... rc4-md5
 2018-01-04 00:11:10 INFO: tcp port reuse enabled
 2018-01-04 00:11:10 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:14:13 INFO: initializing ciphers... aes-128-ctr
 2018-01-04 00:14:16 ERROR: failed to resolve server name, wait 2 seconds
 2018-01-04 00:14:20 ERROR: failed to resolve server name, wait 4 seconds
 2018-01-04 00:14:28 ERROR: failed to resolve server name, wait 8 seconds
 2018-01-04 00:14:44 ERROR: failed to resolve server name, wait 16 seconds
 2018-01-04 00:15:16 ERROR: failed to resolve server name, wait 32 seconds
 2018-01-04 00:16:20 ERROR: failed to resolve server name, wait 64 seconds
 2018-01-04 00:17:53 INFO: initializing ciphers... rc4-md5
 2018-01-04 00:17:53 INFO: tcp port reuse enabled
 2018-01-04 00:17:53 INFO: listening at 127.0.0.1:1086
 2018-01-04 00:29:40 ERROR: remote_recv_cb_send: Broken pipe

**Comment by yangfeicheung** [comment]
看起来像是配置文件信息不太对

**Comment by lawpro** [comment]
换成 Shadowsocksx-NG 1.6.1 试试？

**Comment by bingoku** [comment]
Shadowsocksx-NG 1.7.0 用不了，启动就退出。
Shadowsocksx-NG 1.6.1 可以用的。

**Comment by hiwanz** [comment]
哪里可以手动修改配置文件，已知的配置问题导致启动就崩溃
```
2018-01-06 00:12:37 INFO: initializing ciphers... aes-256-cfb
876  2018-01-06 00:12:37 INFO: tcp port reuse enabled
877  2018-01-06 00:12:37 INFO: listening at 127.0.0.1:1086
878  2018-01-06 00:12:42 ERROR: getpeername: Invalid argument
879  2018-01-06 00:12:42 ERROR: getpeername: Invalid argument
880  2018-01-06 00:13:35 ERROR: getpeername: Invalid argument
881  2018-01-06 00:13:55 ERROR: getpeername: Invalid argument
882  2018-01-06 00:15:57 ERROR: getpeername: Invalid argument
883  2018-01-06 00:16:27 ERROR: getpeername: Invalid argument
884  2018-01-06 00:18:37 ERROR: getpeername: Invalid argument
885  2018-01-06 00:22:17 ERROR: getpeername: Invalid argument
886  2018-01-06 00:22:21 ERROR: getpeername: Invalid argument
887  2018-01-06 00:22:22 ERROR: getpeername: Invalid argument
888  2018-01-06 00:22:22 ERROR: getpeername: Invalid argument
889  2018-01-06 00:22:22 ERROR: getpeername: Invalid argument
890  2018-01-06 00:22:24 ERROR: getpeername: Invalid argument
891  2018-01-06 00:22:24 ERROR: getpeername: Invalid argument
892  2018-01-06 00:22:27 ERROR: getpeername: Invalid argument
893  2018-01-06 00:22:27 ERROR: getpeername: Invalid argument
894  2018-01-06 00:22:28 ERROR: getpeername: Invalid argument
895  2018-01-06 00:22:28 ERROR: getpeername: Invalid argument
896  2018-01-06 00:22:54 ERROR: getpeername: Invalid argument
897  2018-01-06 00:22:54 ERROR: getpeername: Invalid argument
898  2018-01-06 00:22:55 ERROR: getpeername: Invalid argument
899  2018-01-06 00:22:55 ERROR: getpeername: Invalid argument
900  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
901  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
902  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
903  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
904  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
905  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
906  2018-01-06 00:23:05 ERROR: getpeername: Invalid argument
907  2018-01-06 00:23:06 ERROR: getpeername: Invalid argument
908  2018-01-06 00:23:07 ERROR: getpeername: Invalid argument
909  2018-01-06 00:23:08 ERROR: getpeername: Invalid argument
910  2018-01-06 00:23:08 ERROR: getpeername: Invalid argument
911  2018-01-06 00:23:08 ERROR: getpeername: Invalid argument
912  2018-01-06 00:23:09 ERROR: getpeername: Invalid argument
913  2018-01-06 00:23:09 ERROR: getpeername: Invalid argument
914  2018-01-06 00:23:09 ERROR: getpeername: Invalid argument
```

**Comment by hupengbuaa** [comment]
同样是代理服务器配置设置错了(随便设置了一个格式错误的ip)，然后打开就崩溃，卸载后重装依旧不行。请问代理服务器配置信息，保存在哪里，我想直接修改文件里的错误信息。

**Comment by qiuyuzhou** [comment]
It was caused by ping.
Has been fixed in 1.7.1
